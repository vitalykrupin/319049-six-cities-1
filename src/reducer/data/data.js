const initialState = {
  city: {},
  offers: [],
  reviews: [],
  isReviewSending: false,
  didReviewSent: false,
  sendError: null,
};

const getRandomCity = (min, max) => Math.floor(Math.random() * (max - min)) + min;

const snakeToCamel = (word) => word.replace(
    /(_\w)/g,
    (matches) => matches[1].toUpperCase()
);

const normalizeKeys = (obj) => {
  if (Array.isArray(obj)) {
    return obj.map((item) => normalizeKeys(item));
  }

  if (obj !== null && obj.constructor === Object) {
    return Object.keys(obj).reduce((acc, key) => ({
      ...acc,
      [snakeToCamel(key)]: normalizeKeys(obj[key])
    }), {});
  }

  return obj;
};

const changeOffer = (state, newOffer) => {
  return state.offers.map((item) => {
    return item.id === newOffer.id ? newOffer : item;
  });
};

const getCity = (selectedCity, offers) => offers.filter((offer) => offer.city.name === selectedCity)[0].city;

const Operation = {
  loadOffers: () => (dispatch, _getState, api) => {
    return api.get(`/hotels`)
      .then((response) => {
        const preparedData = response.data.map((item) => normalizeKeys(item));
        dispatch(ActionCreator.loadOffers(preparedData));
      });
  },

  loadReviews: (id) => (dispatch, _getState, api) => {
    return api.get(`/comments/${id}`)
      .then((response) => {
        const preparedData = response.data.map((item) => normalizeKeys(item));
        dispatch(ActionCreator.loadReviews(preparedData));
      });
  },

  postReview: (review, id) => (dispatch, _getState, api) => {
    return api.post(`/comments/${id}`, review)
      .then((response) => {
        const preparedData = response.data.map((item) => normalizeKeys(item));
        dispatch(ActionCreator.postReview(preparedData));
        dispatch(ActionCreator.showError(null));
        dispatch(ActionCreator.blockForm(false));
        dispatch(ActionCreator.cleanForm(true));
      })
      .catch((_error) => {
        dispatch(ActionCreator.blockForm(false));
        dispatch(ActionCreator.showError(`Error occured :-(`));
      });
  },
};

const ActionCreator = {
  changeCity: (selectedCity, places) => {
    const city = getCity(selectedCity, places);
    return {
      type: `CHANGE_CITY`,
      payload: city,
    };
  },
  loadOffers: (offers) => ({
    type: `LOAD_OFFERS`,
    payload: offers,
  }),
  loadReviews: (reviews) => ({
    type: `LOAD_REVIEWS`,
    payload: reviews,
  }),
  postReview: (reviews) => ({
    type: `POST_REVIEW`,
    payload: reviews,
  }),
  blockForm: (bool) => ({
    type: `BLOCK_FORM`,
    payload: bool,
  }),
  cleanForm: (bool) => ({
    type: `CLEAN_FORM`,
    payload: bool,
  }),
  showError: (error) => ({
    type: `SHOW_ERROR`,
    payload: error,
  }),
  addToFavorites: (data) => ({
    type: `ADD_TO_FAVORITES`,
    payload: data,
  }),
  deleteFromFavorites: (data) => ({
    type: `DELETE_FROM_FAVORITES`,
    payload: data,
  }),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `CHANGE_CITY`: return Object.assign({}, state, {
      city: action.payload,
    });
    case `LOAD_OFFERS`: return Object.assign({}, state, {
      city: action.payload[getRandomCity(1, action.payload.length)].city,
      offers: action.payload,
    });
    case `LOAD_REVIEWS`: return Object.assign({}, state, {
      reviews: action.payload,
    });
    case `POST_REVIEW`: return Object.assign({}, state, {
      reviews: action.payload,
    });
    case `BLOCK_FORM`: return Object.assign({}, state, {
      isReviewSending: action.payload,
    });
    case `CLEAN_FORM`: return Object.assign({}, state, {
      didReviewSent: action.payload,
    });
    case `SHOW_ERROR`: return Object.assign({}, state, {
      sendError: action.payload,
    });
    case `ADD_TO_FAVORITES`: return Object.assign({}, state, {
      offers: changeOffer(state, action.payload),
    });
    case `DELETE_FROM_FAVORITES`: return Object.assign({}, state, {
      offers: changeOffer(state, action.payload),
    });
  }
  return state;
};

export {reducer, ActionCreator, Operation, normalizeKeys};
