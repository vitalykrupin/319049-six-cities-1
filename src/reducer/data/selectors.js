import NameSpace from '../name-space';

const NAME_SPACE = NameSpace.DATA;

export const getOffers = (state) => {
  return state[NAME_SPACE].offers;
};

export const getCity = (state) => {
  return state[NAME_SPACE].city;
};

export const getReviews = (state) => {
  return state[NAME_SPACE].reviews;
};

export const getIsReviewSending = (state) => {
  return state[NAME_SPACE].isReviewSending;
};

export const getDidReviewSent = (state) => {
  return state[NAME_SPACE].didReviewSent;
};

export const getError = (state) => {
  return state[NAME_SPACE].sendError;
};
