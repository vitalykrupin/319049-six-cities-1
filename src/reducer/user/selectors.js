import NameSpace from '../name-space';

const NAME_SPACE = NameSpace.USER;

export const getAuthorizationStatus = (state) => {
  return state[NAME_SPACE].isAuthorizationRequired;
};

export const getAuthError = (state) => {
  return state[NAME_SPACE].authError && state[NAME_SPACE].authError;
};

export const getUserData = (state) => {
  return state[NAME_SPACE].user;
};

export const getFavorites = (state) => {
  return state[NAME_SPACE].favorites;
};
