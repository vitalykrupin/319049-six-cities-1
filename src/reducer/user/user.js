import {RequestStatus} from '../../constants';
import {
  ActionCreator,
  normalizeKeys,
  Operation as DataOperation
} from '../data/data';

const initialState = {
  isAuthorizationRequired: true,
  user: {},
  authError: null,
  favorites: null,
};

const Operation = {
  logIn: () => (dispatch, _getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        const email = response.data.email;
        const password = null;
        api.post(`/login`, {email, password})
        .then((resp) => {
          if (resp.data) {
            dispatch(UserActionCreator.requireAuthorization(false));
            dispatch(UserActionCreator.authorizeUser(resp.data));
            dispatch(UserActionCreator.authError(null));
            dispatch(DataOperation.loadOffers());
          }
        })
        .catch((error) => {
          if (error.response.status) {
            if (error.response.status === RequestStatus.BAD_REQUEST) {
              dispatch(UserActionCreator.authError(error.response.data.error));
              throw error;
            }
          }
        });
      })
      .catch((error) => {
        if (error.response.status === RequestStatus.FORBIDDEN || error.response.status === RequestStatus.UNAUTHORIZED) {
          dispatch(UserActionCreator.logOut());
        }
      });
  },

  authorizeUser: (email, password) => (dispatch, _getState, api) => {
    return api.post(`/login`, {email, password})
      .then((response) => {
        if (response.data) {
          dispatch(UserActionCreator.requireAuthorization(false));
          dispatch(UserActionCreator.authorizeUser(response.data));
          dispatch(UserActionCreator.authError(null));
          dispatch(DataOperation.loadOffers());
        }
      })
      .catch((error) => {
        if (error.response.status) {
          if (error.response.status === RequestStatus.BAD_REQUEST) {
            dispatch(UserActionCreator.authError(error.response.data.error));
          }
        }
      });
  },

  loadFavorites: () => (dispatch, _getState, api) => {
    return api.get(`/favorite`)
      .then((response) => {
        const preparedData = response.data.map((item) => normalizeKeys(item));
        dispatch(UserActionCreator.loadFavorites(preparedData));
      });
  },

  addToFavorites: (id) => (dispatch, _getState, api) => {
    return api.post(`/favorite/${id}/1`)
      .then((response) => {
        const preparedData = normalizeKeys(response.data);
        dispatch(ActionCreator.addToFavorites(preparedData));
      })
      .catch((_error) => {});
  },

  deleteFromFavorites: (id) => (dispatch, _getState, api) => {
    return api.post(`/favorite/${id}/0`)
      .then((response) => {
        const preparedData = normalizeKeys(response.data);
        dispatch(ActionCreator.deleteFromFavorites(preparedData));
      })
      .catch((_error) => {});
  },
};

const UserActionCreator = {
  requireAuthorization: (status) => ({
    type: `REQUIRE_AUTHORIZATION`,
    payload: status,
  }),

  authorizeUser: (user) => ({
    type: `AUTHORIZE_USER`,
    payload: normalizeKeys(user),
  }),

  logOut: () => ({
    type: `LOG_OUT`,
  }),

  authError: (error) => ({
    type: `AUTH_ERROR`,
    payload: error,
  }),

  loadFavorites: (favorites) => ({
    type: `LOAD_FAVORITES`,
    payload: favorites,
  }),
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `REQUIRE_AUTHORIZATION`: return Object.assign({}, state, {
      isAuthorizationRequired: action.payload,
    });
    case `AUTHORIZE_USER`: return Object.assign({}, state, {
      user: action.payload,
    });
    case `LOG_OUT`: return Object.assign({}, state, {
      user: {},
      isAuthorizationRequired: true,
    });
    case `AUTH_ERROR`: return Object.assign({}, state, {
      authError: action.payload,
    });
    case `LOAD_FAVORITES`: return Object.assign({}, state, {
      favorites: action.payload,
    });
  }
  return state;
};

export {reducer, UserActionCreator, Operation};
