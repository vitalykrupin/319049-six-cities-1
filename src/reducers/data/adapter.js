export const adapter = (data) => {
  return data.map((place) => {
    return {
      bedrooms: place.bedrooms,
      city: {
        location: [place.city.location.latitude, place.city.location.longitude],
        name: place.city.name
      },
      description: place.description,
      goods: place.goods,
      host: {
        avatarURL: place.host.avatar_url,
        id: place.host.id,
        isPro: place.host.is_pro,
        name: place.host.name
      },
      id: place.id,
      images: place.images,
      isFavorite: place.is_favorite,
      isPremium: place.isPremium,
      location: [place.location.latitude, place.location.longitude],
      maxAdults: place.max_adults,
      previewImage: place.preview_image,
      price: place.price,
      rating: place.rating,
      title: place.title,
      type: place.type
    };
  });
};
