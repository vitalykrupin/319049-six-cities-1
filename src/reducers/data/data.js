const initialState = {
  fetchedOffers: []
};

const actionsType = {
  FETCH_OFFERS: `FETCH_OFFERS`
};

const ActionCreators = {
  fetchOffers: (offers) => ({
    type: `FETCH_OFFERS`,
    payload: offers
  })
};

const Operations = {
  fetchOffers: () => (dispatch, getState, api) => {
    return api.get(`/hotels`)
      .then((response) => {
        dispatch(ActionCreators.fetchOffers(response.data));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actionsType.FETCH_OFFERS: return Object.assign({}, state, {
      fetchedOffers: action.payload
    });
  }

  return state;
};

export {reducer, ActionCreators, Operations, actionsType};
