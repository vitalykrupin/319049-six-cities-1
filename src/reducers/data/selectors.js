import {createSelector} from 'reselect';
import {NameSpace} from '../namespaces';
import {SortType} from '../../utils';

const NAMESPACE = NameSpace.DATA;
const MAX_CITIES = 6;

const sortCitiesByName = (offers) => {
  const cities = offers.map((offer) => offer.city).sort((a, b) => {
    const nameA = a.name.toUpperCase();
    const nameB = b.name.toUpperCase();

    if (nameA < nameB) {
      return -1;
    } else if (nameA > nameB) {
      return 1;
    } else {
      return 0;
    }
  });

  return cities.reduce((prev, current) => {
    if (!prev.includes(current)) {
      prev.push(current);
    }
    return prev;
  }, []);
};

const getUnicCities = (arr) => {
  const unicCities = [];

  for (let i = 0; i < arr.length; i++) {
    if (i === 0 || arr[i].name !== arr[i - 1].name) {
      unicCities.push(arr[i]);
    }
  }

  return unicCities;
};

const getRandomCityOffer = (offers) => {
  const min = 0;
  const max = Math.floor(offers.length);
  return offers[Math.floor(Math.random() * (max - min)) + min];
};

export const sortFavoritesListByCities = (offers) => {
  const cities = {};

  offers.forEach((item) => {
    if (!cities[item.city.name]) {
      cities[item.city.name] = [];
    }
  });

  offers.forEach((item) => {
    const key = item.city.name;
    if (cities[key]) {
      cities[key].push(item);
    }
  });

  return cities;
};

export const getOffers = (state) => {
  return state[NAMESPACE].rentalOffers;
};

export const getCurrentCity = (state) => {
  return state[NAMESPACE].currentCity;
};

export const getSortValue = (state) => {
  return state[NAMESPACE].sortValue;
};

export const getOffersLoadStatus = (state) => {
  return state[NAMESPACE].offersLoaded;
};

export const getCities = createSelector(
    [getOffers],
    (offers) => getUnicCities(sortCitiesByName(offers)).slice(0, MAX_CITIES)
);

export const getCityOffers = createSelector(
    getOffers,
    getCurrentCity,
    (offers, city) => offers.filter((offer) => offer.city.name === city.name)
);

export const getRandomOffer = createSelector(
    [getOffers],
    (state) => getRandomCityOffer(state)
);

export const sortOffers = createSelector(
    getCityOffers,
    getSortValue,
    (offers, state) => {
      switch (state) {
        case (SortType.HIGH_TO_LOW):
          return offers.sort((a, b) => b.price - a.price);
        case (SortType.LOW_TO_HIGH):
          return offers.sort((a, b) => a.price - b.price);
        case (SortType.TOP_RATED):
          return offers.sort((a, b) => b.rating - a.rating);
        default:
          return offers.sort((a, b) => a.id - b.id);
      }
    }
);

export const getOfferById = (state, id) =>
  getOffers(state).filter((item) => item.id === +id)[0];

export const getCloserOffers = (state, id) => {
  const offer = getOfferById(state, id);

  return getOffers(state)
    .filter((item) => item.city.name === offer.city.name)
    .filter((item) => item.id !== +id);
};

export const getFavoriteOffers = (state) =>
  getOffers(state).filter((item) => item.is_favorite);
