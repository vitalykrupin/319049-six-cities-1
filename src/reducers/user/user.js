const initialState = {
  isAuthenticated: false,
  pendingAuthorization: true,
  user: {},
};

const ActionType = {
  PENDING_AUTHORIZATION: `PENDING_AUTHORIZATION`,
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  LOGIN: `LOGIN`,
};

const ActionCreator = ({
  pendingAuthorization: (status) => ({
    type: ActionType.PENDING_AUTHORIZATION,
    payload: status,
  }),

  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  }),

  login: (user) => ({
    type: ActionType.LOGIN,
    payload: user,
  })
});

const Operation = {
  login: (data) => (dispatch, getState, api) => {
    return api.post(`/login`, data)
      .then((response) => {
        if (response.data) {
          dispatch(ActionCreator.login(response.data));
          dispatch(ActionCreator.requireAuthorization(true));
        }
      })
      .finally(() => {
        dispatch(ActionCreator.pendingAuthorization(false));
      });
  },

  checkAuthorization: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((response) => {
        if (response.data) {
          dispatch(ActionCreator.requireAuthorization(true));
          dispatch(ActionCreator.login(response.data));
        }
      })
      .finally(() => {
        dispatch(ActionCreator.pendingAuthorization(false));
      });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return Object.assign({}, state, {
        isAuthenticated: action.payload,
      });
    case ActionType.LOGIN:
      return Object.assign({}, state, {
        user: action.payload,
      });
    case ActionType.PENDING_AUTHORIZATION:
      return Object.assign({}, state, {
        pendingAuthorization: action.payload,
      });
  }

  return state;
};

export {reducer, ActionCreator, Operation, ActionType};
