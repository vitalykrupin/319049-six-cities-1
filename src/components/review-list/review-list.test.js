import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ReviewList from '../review-list/review-list';
import {Operation} from '../../reducer/data/data';
import {Operation as OperationUser} from '../../reducer/user/user';
import NameSpace from '../../reducer/name-space';

jest.mock(`../../reducer/data/data`);
jest.mock(`../../reducer/user/user`);
Operation.loadReviews = () => (dispatch) => dispatch(jest.fn());
OperationUser.authorizeUser = () => (dispatch) => dispatch(jest.fn());

const NAME_SPACE = NameSpace.DATA;
const NAME_SPACE_USER = NameSpace.USER;
const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {};
initialState[NAME_SPACE] = {
  city: {},
  offers: [],
  reviews: [
    {
      id: 1,
      comment: `Weird place`,
      rating: 1.5,
      date: `2019-05-16T21:02:58.227Z`,
      user: {
        avatarUrl: `path.jpg`,
        id: 8,
        isPro: false,
        name: `Kurt`,
      },
    },
    {
      id: 2,
      comment: `Strange place`,
      rating: 2.5,
      date: `2019-06-16T21:02:58.227Z`,
      user: {
        avatarUrl: `path.jpg`,
        id: 9,
        isPro: false,
        name: `Kate`,
      },
    },
  ],
  isReviewSending: false,
  didReviewSent: false,
  sendError: null,
};
initialState[NAME_SPACE_USER] = {
  user: {},
  authError: null,
  isAuthorizationRequired: false,
};
const store = mockStore(initialState);

describe(`ReviewList`, () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <ReviewList id={1} />
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
