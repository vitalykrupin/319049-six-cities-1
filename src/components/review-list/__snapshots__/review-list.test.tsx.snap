// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList renders correctly 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      2
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="path.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Kate
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "50%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Strange place
        </p>
        <time
          className="reviews__time"
          dateTime="2019-06-16T21:02:58.227Z"
        >
          June
           
          2019
        </time>
      </div>
    </li>
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="path.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Kurt
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "30%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          Weird place
        </p>
        <time
          className="reviews__time"
          dateTime="2019-05-16T21:02:58.227Z"
        >
          May
           
          2019
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
    onSubmit={[Function]}
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="5-stars"
        name="rating"
        onClick={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="4-stars"
        name="rating"
        onClick={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="3-stars"
        name="rating"
        onClick={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="2-stars"
        name="rating"
        onClick={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        className="form__rating-input visually-hidden"
        disabled={false}
        id="1-star"
        name="rating"
        onClick={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-star"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      disabled={false}
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        style={
          Object {
            "color": "red",
            "fontSize": "20px",
          }
        }
      >
        
      </p>
      <p
        className="reviews__help"
      >
        To submit review please make sure to set 
        <span
          className="reviews__star"
        >
          rating
        </span>
         and describe your stay with at least 
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
